images/vault/tests/main.tf

```diff
@@ -17,54 +18,69 @@ data "oci_string" "ref" {
   input    = each.value
 }
 
-resource "random_pet" "suffix" {}
+data "imagetest_inventory" "this" {}
 
-resource "helm_release" "vault" {
-  name             = "vault-${random_pet.suffix.id}"
-  namespace        = "vault-system-${random_pet.suffix.id}"
-  repository       = "https://helm.releases.hashicorp.com"
-  chart            = "vault"
-  create_namespace = true
+resource "imagetest_harness_k3s" "this" {
+  name      = "vault"
+  inventory = data.imagetest_inventory.this
+}
+
+module "helm" {
+  source = "../../../tflib/imagetest/helm"
 
-  values = [
-    jsonencode({
-      injector = {
-        agentImage = {
-          repository = data.oci_string.ref["vault"].registry_repo
-          tag        = data.oci_string.ref["vault"].pseudo_tag
-        }
-        image = {
-          repository = data.oci_string.ref["vault-k8s"].registry_repo
-          tag        = data.oci_string.ref["vault-k8s"].pseudo_tag
-        }
+  name      = "vault"
+  namespace = "vault-system"
+  chart     = "vault"
+  repo      = "https://helm.releases.hashicorp.com"
+
+  values = {
+    injector = {
+      agentImage = {
+        repository = data.oci_string.ref["vault"].registry_repo
+        tag        = data.oci_string.ref["vault"].pseudo_tag
       }
-      server = {
-        image = {
-          repository = data.oci_string.ref["vault"].registry_repo
-          tag        = data.oci_string.ref["vault"].pseudo_tag
-        }
+      image = {
+        repository = data.oci_string.ref["vault-k8s"].registry_repo
+        tag        = data.oci_string.ref["vault-k8s"].pseudo_tag
       }
-    })
-  ]
-}
-
-data "oci_exec_test" "acceptance" {
-  digest     = var.digests["vault"]
-  script     = "${path.module}/acceptance.sh"
-  depends_on = [helm_release.vault]
-  env {
-    name  = "NAME"
-    value = "vault-${random_pet.suffix.id}"
-  }
-  env {
-    name  = "NAMESPACE"
-    value = "vault-system-${random_pet.suffix.id}"
+    }
+    server = {
+      image = {
+        repository = data.oci_string.ref["vault"].registry_repo
+        tag        = data.oci_string.ref["vault"].pseudo_tag
+      }
+    }
   }
 }
 
-module "helm_cleanup" {
-  depends_on = [data.oci_exec_test.acceptance]
-  source     = "../../../tflib/helm-cleanup"
-  name       = helm_release.vault.id
-  namespace  = helm_release.vault.namespace
+resource "imagetest_feature" "basic" {
+  harness     = imagetest_harness_k3s.this
+  name        = "Basic"
+  description = "Basic functionality of the vault image."
+
+  steps = [
+    {
+      name = "Helm install vault"
+      cmd  = module.helm.install_cmd
+    },
+    {
+      name = "Acceptance test"
+      cmd  = <<EOF
```

Comment:
This is some weird stuff. What processes this '<<EOF' ?
If it is passed to a shell, then you will get interpretation inside the body.
Ie, the 'kubectl exec' command will run _on the host_ and the output of that placed into the rendered document.

```bash
$ FOO=outside sh -c 'cat <<EOF
FOO=inside
echo $(echo $FOO)
EOF'

FOO=inside
echo outside
```

If you did it with double quotes around the EOF, then you get what you were after:

```bash
$ FOO=outside sh -c 'cat <<"EOF"
FOO=inside
echo $(echo $FOO)
EOF'

FOO=inside
echo $(echo $FOO)
```

Reply:
or is '<<' a terraform template operator? link?

Reply:
https://developer.hashicorp.com/terraform/language/expressions/strings#heredoc-strings

Reply:
`<<` is terraforms multiline string: https://developer.hashicorp.com/terraform/language/expressions/strings#heredoc-strings

this gets passed in as a raw string to `/bin/sh -c '...'`